app/src/main/java/com/oracle/javafx/scenebuilder/app/welcomedialog/WelcomeDialogWindowController.java

```diff
@@ -101,38 +102,51 @@ protected void controllerDidLoadFxml() {
         super.controllerDidLoadFxml();
         assert recentDocuments != null;
 
-        PreferencesRecordGlobal preferencesRecordGlobal = PreferencesController
-                .getSingleton().getRecordGlobal();
-        List<String> recentItems = preferencesRecordGlobal.getRecentItems();
-        if (recentItems.size() == 0) {
-            Label noRecentItems = new Label(I18N.getString("welcome.recent.items.no.recent.items"));
-            noRecentItems.getStyleClass().add("no-recent-items-label");
-            recentDocuments.getChildren().add(noRecentItems);
-        }
-        for (int row = 0; row < preferencesRecordGlobal.getRecentItemsSize(); ++row) {
-            if (recentItems.size() < row + 1) {
-                break;
-            }
-
-            String recentItem = recentItems.get(row);
-            File recentItemFile = new File(recentItems.get(row));
-            String recentItemTitle = recentItemFile.getName();
-            Button recentDocument = new Button(recentItemTitle);
-            recentDocument.getStyleClass().add("recent-document");
-            recentDocument.setMaxWidth(Double.MAX_VALUE);
-            recentDocument.setAlignment(Pos.BASELINE_LEFT);
-            recentDocuments.getChildren().add(recentDocument);
-
-            recentDocument.setOnAction(event -> fireOpenRecentProject(event, recentItem));
-            recentDocument.setTooltip(new Tooltip(recentItem));
-        }
+        loadAndPopulateRecentItemsInBackground();
 
         emptyApp.setUserData(Template.EMPTY_APP);
 
         setOnTemplateChosen(sceneBuilderApp::performNewTemplate);
         setupTemplateButtonHandlers();
     }
 
+    private void loadAndPopulateRecentItemsInBackground() {
+        new Thread(() -> {
+            PreferencesRecordGlobal preferencesRecordGlobal = PreferencesController
+                    .getSingleton().getRecordGlobal();
+            List<String> recentItems = preferencesRecordGlobal.getRecentItems();
+
+            if (recentItems.size() == 0) {
+                Label noRecentItems = new Label(I18N.getString("welcome.recent.items.no.recent.items"));
+                noRecentItems.getStyleClass().add("no-recent-items-label");
+
+                Platform.runLater(() -> {
+                    recentDocuments.getChildren().add(noRecentItems);
+                });
+            }
+
+            for (int row = 0; row < preferencesRecordGlobal.getRecentItemsSize(); ++row) {

```

Comment:
Could be considered syntactic preference, but this may be cleaner and reduces the size of the loadAndPopulateRecentItemsInBackground method.

```Java
IntStream.range(0, preferencesRecordGlobal.getRecentItemsSize())
    .mapToObj(i -> recentItems.get(i))
    .forEach(this::addRecentDocument);
```

```Java
private void addRecentDocument(String recentItem) {
    File recentItemFile = new File(recentItem);
    String recentItemTitle = recentItemFile.getName();
    Button recentDocument = new Button(recentItemTitle);
    recentDocument.getStyleClass().add("recent-document");
    recentDocument.setMaxWidth(Double.MAX_VALUE);
    recentDocument.setAlignment(Pos.BASELINE_LEFT);
    recentDocument.setOnAction(event -> fireOpenRecentProject(event, recentItem));
    recentDocument.setTooltip(new Tooltip(recentItem));

    Platform.runLater(
         () -> {
            recentDocuments.getChildren().add(recentDocument);
        });
    }
```

Reply:
Looks good, will refactor in a future PR, but not this one, for easier review.